### ISSUE TITLE
Support for variable length timeseries

### ISSUE THREAD

**Issue by garethmd** [jdb78/pytorch-forecasting#1364]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0.1
- Python version: 3.8
- Operating System: OSX

I'd like to know if it is possible to use Timeseries with variable length sequences and if so how would I go about preparing the dataset. For example if I have a single dataframe detailing sales figures for products in a portfolio, but each product was introduced at different points in time I may have a timeseries for product A from 2019-01-01 to present and a timeseries for product B from 2021-03-01 to present. 

When I have done this sort of thing in the past I have added padding ``nn.utils.rnn.pad_sequence`` to the dataset or dataloader in order to produce a data batch and then the model and loss function would have to handle the padding values accordingly. I couldn't see if this was supported and I'd be grateful for some clarification.
